### PR TITLE
Testing/3 30

### PR DESCRIPTION
**Pull request by KianChilds**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/shooterSubsystem/ShooterCommandFactory.java [hunk 1]
```diff
@@ -82,6 +86,13 @@ public Command shooterIdle() {
         subsystem
             .run(
                 () -> {
+                  if (RSM.getPlacementMode() == PlacementMode.SPEAKER) {
+                    subsystem.setShooterSpeeds(
+                        new SpeakerConfig(
+                            -1,
+                            ShooterConstants.SHOOTER_IDLE_SPEED_RIGHT,
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/shooterSubsystem/ShooterCommandFactory.java:93]: IDLE isn't the right speed for a speaker shot.  It's showing as 400 RPM

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java [hunk 2]
```diff
@@ -160,7 +201,7 @@ public Command longRangeShot() {
                 visionTracking();
               }
               if (Constants.enabledSubsystems.turretPitchEnabled) {
-                double distance = distanceEstimateMeters();
+                double distance = distanceEstimateMeters() - 0.3;
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java:204]: let's put 0.3 as a constant

File: src/main/java/frc/robot/utilities/DebugEntry.java [hunk 3]
```diff
@@ -42,6 +43,8 @@ public DebugEntry(T defaultValue, String name, Subsystem subsystem) {
       localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "boolean");
       localConsumer = (a) -> datalog.appendBoolean(localEntry, (Boolean) a, 0);
 
+    } else if (defaultValue instanceof ShooterMode) {
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/utilities/DebugEntry.java:46]: We are passing in strings for shooterModeLog, not instances of ShooterMode.

Let's remove this workaround, and think about how to do this in a more elegant way

**Review by caseyjbrotherton — CHANGES_REQUESTED** [review]
changes in comments

File: src/main/java/frc/robot/subsystems/shooterSubsystem/ShooterCommandFactory.java (hunk 1 above)
Comment (ModBoyex) [reply src/main/java/frc/robot/subsystems/shooterSubsystem/ShooterCommandFactory.java:93]: This is the idle mode. The check for mode earlier is so we are only idling when we are about to shot

**Review by caseyjbrotherton — APPROVED** [review]

**Pull request merged**